tests/test_components/test_config_component.py

```diff
@@ -11,6 +11,12 @@
 from hydromt_wflow.utils import DATADIR
 
 
+@pytest.fixture
+def mock_model(mock_model_factory) -> MagicMock:
+    """Create a mock model with a root."""
+    return mock_model_factory(mode="w")
```

Comment:
I just merged states so maybe you'll need to add it in test_states_component too ?